jetty-core/jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/StdErrAppender.java

```diff
@@ -199,9 +202,16 @@ private String renderedLevel(Level level)
         }
     }
 
-    private void appendCause(StringBuilder builder, Throwable cause, String indent)
+    private void appendCause(StringBuilder builder, Throwable cause, String indent, Set<Throwable> visited)
     {
         builder.append(EOL).append(indent);
+        if (visited.contains(cause))
+        {
+            builder.append("[CIRCULAR REFERENCE: ").append(cause).append("]");
```

Comment:
`append(cause)` appends a naked `cause.toString()`, which in other places is sanitized by calling `appendEscaped()`.

Reply:
Changed.